LEGO1/mxdsobject.cpp

```diff
@@ -0,0 +1,25 @@
+#include "mxdsobject.h"
+
+#include <string.h>
+#include <stdlib.h>
+
+void MxDSObject::SetObjectName(const char *p_name)
+{
+  // TODO: instead of the expected CMP EAX,ESI we get CMP ESI,EAX
+  if (p_name != this->m_name)
```

Comment:
Does the order of comparison matter? `p_name` <-> `this->m_name`?

Reply:
Unfortunately makes no difference, that's the first thing I tried

Reply:
What about swapping it to compare for equality + early return? Not that it actually matters in the least, functionally, just to exhaust all potential options.

Reply:
Opinions on using inline assembly?

Reply:
My 2c: only use (inline) assembly in performance critical locations, not for a simple if.

Reply:
Thanks guys, I'm not sure when it happened but now it's matching:

`MxDSObject::SetObjectName (0x100bf8e0 / 0x10001110) is 100.00% similar to the original`

We still have that same issue with `MxString::operator=` though:

`MxString::operator= (0x100ae510 / 0x100021d0) is 97.73% similar to the original`